src/components/ChatEntry.jsx

```diff
@@ -1,20 +1,56 @@
 import './ChatEntry.css';
+import './TimeStamp';
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+import TimeStamp from './TimeStamp';
 
-const ChatEntry = () => {
-  return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
-      <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+const ChatEntry = ({ id, body, sender, timeStamp, liked, onclickChat }) => {
+  // not using props as need to destructure and forward the onclickChat function properly.
+  const [isLiked, setIsLiked] = useState(liked);
+
+  const onLikeClick = () => {
+    // Toggle the like state
+    setIsLiked(!isLiked);
+    onclickChat(id);
+    //we are passing these states to ChatLog ->App.jsx
+  };
+  return sender === 'Vladimir'? (
+    <div className='chat-entry local'>
+      <h2 className='entry-name'>{sender}</h2>
+      <section className='entry-bubble'>
+        <p>{body}</p>
+        <p className='entry-time'>
+          <TimeStamp time={timeStamp} />
+        </p>
+        <button className='like' onClick={onLikeClick}>
+          {isLiked ? '❤️' : '🤍'}
+        </button>
+      </section>
+    </div>
+  ) : (
+    <div className='chat-entry remote'>
+      <h2 className='entry-name'>{sender}</h2>
+      <section className='entry-bubble'>
+        <p>{body}</p>
+        <p className='entry-time'>
+          <TimeStamp time={timeStamp} />
+        </p>
+        <button className='like' onClick={onLikeClick}>
+          {isLiked ? '❤️' : '🤍'}
+        </button>
```

Comment:
This section duplicates a lot of code.

Instead of using a ternary operator here to decide which set of HTML elements should be returned, how can you conditionally set variables and use those variables to dynamically set classes, etc?

For example:
```js
const sender ===  Vladimir' ? 'local' : 'remote';
<div className='chat-entry {sender}'>
```